stf.core/scripts/stf.pl

```diff
@@ -162,20 +162,41 @@
 my $timestamp = sprintf '%04d%02d%02d-%02d%02d%02d', $year+1900, $mon+1, $mday, $hour, $min, $sec;
 
 # Work out where the working, temp and results directories are going
-my $results_root = stfArguments::get_and_check_argument("results-root");
+my $results_root  = stfArguments::get_and_check_argument("results-root");
+my $retain_number = stfArguments::get_and_check_argument("retain");
+my $retain_limit  = stfArguments::get_and_check_argument("retain-limit");
 my $test_dir = $results_root . "/$timestamp-$test_name";
 my $debug_dir   = $test_dir . "/debug";
 my $results_dir = $test_dir . "/results";
 my $generation_dir = $test_dir . "/generation";
 my $setup_dir      = $test_dir . "/setUp";
 my $execute_dir    = $test_dir . "/execute";
 my $teardown_dir   = $test_dir . "/tearDown";
+my $delete_results_root = $FALSE;
 
 my @results_dirs;
 
+if ( length $retain_limit ) {
+   $results_retention_limit = $retain_limit;
+}
+
+if ( length $retain_number ) {
+   if ( $retain_number == 0 ) {
```

Comment:
Personally I'd prefer not having the magic number 0 meaning 1000000.  Apart from being cryptic it means you can't specify 0 to mean 'keep no results'.

Reply:
Fair enough - I did also consider using -1 but happy to just remove the clause altogether